tests/core/policies/test_unexpected_intent_policy.py

```diff
@@ -0,0 +1,826 @@
+        is_unlikely: bool,
+        tmp_path: Path,
+    ):
+        loaded_policy = self.persist_and_load_policy(trained_policy, tmp_path)
+        # Construct dummy similarities
+        similarities = np.array([[0.0] * len(default_domain.intents)])
+        dummy_intent_index = 4
+        similarities[0, dummy_intent_index] = predicted_similarity
+
+        loaded_policy.label_thresholds[dummy_intent_index] = threshold_value
+        query_intent = default_domain.intents[dummy_intent_index]
+
+        unlikely_intent_prediction = loaded_policy._check_unlikely_intent(
+            default_domain, similarities, query_intent
+        )
+
+        assert is_unlikely == unlikely_intent_prediction
+
+    def test_should_check_for_intent(
+        self,
+        trained_policy: UnexpecTEDIntentPolicy,
+        default_domain: Domain,
+        tmp_path: Path,
+    ):
+        loaded_policy = self.persist_and_load_policy(trained_policy, tmp_path)
+
+        intent_index = 0
+        assert (
+            loaded_policy._should_check_for_intent(
+                default_domain.intents[intent_index], default_domain
+            )
+            is False
+        )
+
+        intent_index = 4
+        assert loaded_policy._should_check_for_intent(
+            default_domain.intents[intent_index], default_domain
+        )
+
+        loaded_policy.config[IGNORE_INTENTS_LIST] = [
+            default_domain.intents[intent_index]
+        ]
+        assert (
+            loaded_policy._should_check_for_intent(
+                default_domain.intents[intent_index], default_domain
+            )
+            is False
+        )
+
+    def test_no_action_unlikely_intent_prediction(
+        self,
+        trained_policy: UnexpecTEDIntentPolicy,
+        default_domain: Domain,
+        tmp_path: Path,
+    ):
+        loaded_policy = self.persist_and_load_policy(trained_policy, tmp_path)
+
+        expected_probabilities = [0] * default_domain.num_actions
+
+        interpreter = RegexInterpreter()
+        tracker = DialogueStateTracker(sender_id="init", slots=default_domain.slots)
+        prediction = loaded_policy.predict_action_probabilities(
+            tracker, default_domain, interpreter
+        )
+
+        assert prediction.probabilities == expected_probabilities
+
+        tracker.update_with_events(
+            [
+                UserUttered(text="hello", intent={"name": "greet"}),
+                ActionExecuted(action_name="utter_greet"),
+            ],
+            default_domain,
+        )
+        prediction = loaded_policy.predict_action_probabilities(
+            tracker, default_domain, interpreter
+        )
+
+        assert prediction.probabilities == expected_probabilities
+
+        loaded_policy.model = None
+
+        prediction = loaded_policy.predict_action_probabilities(
+            tracker, default_domain, interpreter
+        )
+
+        assert prediction.probabilities == expected_probabilities
```

Comment:
What is this case supposed to represent? What kind of real scenario?

Reply:
It's supposed to test [this](https://github.com/RasaHQ/rasa/blob/19795211dd5e05d22ca00d3b4b70480bc01a7822/rasa/core/policies/unexpected_intent_policy.py#L498) `if` construct. When there is no data for the policy to be trained but still included in the config, the policy returns an instance of itself with an empty model. During prediction, the policy should return the default predictions in this case.

Reply:
I see, thanks Daksh!